### ISSUE TITLE
1.1.373 issue - Expected class but received "Literal[0]" & Too few type arguments provided for "FixtureFunction"

### ISSUE THREAD

**Issue by lexicalunit** [microsoft/pyright#8607]
**Describe the bug**
In version `1.1.373` I'm getting a bunch of cryptic errors with code that passed fine in version `1.1.372`.

**Code or Screenshots**

I was able to create a minimal example:

```python
from sqlalchemy.engine import create_engine
from sqlalchemy.orm import sessionmaker

database_url = "whatever:///does/not/matter"
engine = create_engine(database_url)
connection = engine.connect()
session_maker = sessionmaker(bind=connection)
session = session_maker()
rows = session.execute("SELECT xid FROM guilds").fetchall()
xids = [int(row[0]) for row in rows]
```

You can see the full error logs in my github actions workflow: https://github.com/lexicalunit/spellbot/actions/runs/10168666850/job/28123798890

**VS Code extension or command-line**
- [x] Command Line

```sh
➜ poetry run pyright tests src example.py
/Users/me/source/lexicalunit/spellbot/example.py
  /Users/me/source/lexicalunit/spellbot/example.py:10:17 - error: Expected class but received "Literal[0]" (reportGeneralTypeIssues)
  /Users/me/source/lexicalunit/spellbot/example.py:10:17 - error: Too few type arguments provided for "FixtureFunction[SimpleFixtureFunction@FixtureFunction, FactoryFixtureFunction@FixtureFunction]"; expected 2 but received 1 (reportInvalidTypeForm)
```

However, if I run `pyright` on a single file, I get no error:

```sh
➜ poetry run pyright example.py
0 errors, 0 warnings, 0 informations
```

It only happens when I execute `pyright` on all my files.

**Additional Information**

```sh
➜ poetry run pip freeze | grep -iE "sqlalchemy|pyright"
pyright==1.1.373
SQLAlchemy==2.0.31
sqlalchemy-stubs==0.4
SQLAlchemy-Utils==0.41.2

➜ poetry run python --version
Python 3.12.4
```

**Comment by erictraut** [comment]
I'm not able to repro what you're seeing. If I install the "sqlalchemy-stubs` package, I see no errors from pyright. I'm guessing that you have something else installed in your venv that's affecting the results. Are you able to repro the problem with a clean venv?

My understanding is that sqlalchemy 2.x is fully typed — that is, it has inline type information — and that it doesn't require the use of an external stub library. By adding the stub library, you're overriding the type information provided by the library maintainers with outdated and incomplete type information. I think the stub library is intended for use with sqlalchemy 1.x, which didn't have type information. You might want to try removing it from your environment. If I don't install `sqlalchemy-stubs`, I receive a type error for the `session.execute` call because this call apparently doesn't accept a `str` argument.

**Comment by erictraut** [comment]
Moving to a discussion topic because I don't think this is a bug in pyright.
